qiskit/qpy/binary_io/circuits.py

```diff
@@ -252,7 +252,7 @@ def _read_instruction(file_obj, circuit, registers, custom_operations, version,
     if gate_name in {"IfElseOp", "WhileLoopOp"}:
         gate = gate_class(condition_tuple, *params)
     elif version >= 5 and issubclass(gate_class, ControlledGate):
-        if gate_name in {"MCPhaseGate", "MCU1Gate"}:
+        if gate_name in {"MCPhaseGate", "MCU1Gate", "MCXGrayCode"}:
```

Comment:
I'm just curious if there are other gates in the mcx family that have the same constructor and need this. I always get lost in all the mcx variants. But it might be good to check we've got all of them covered here

Reply:
You were right @mtreinish, qpy also fails for other MCX gates. Namely (I think these are all):
```
MCXRecursive
MCXVChain
MCXGate
```
I can add these to the PR. I though I had tested it before realizing that with 2 control qubits, the gate ends up being of type `CCXGate`, so of course all my checks passed.